trio_run_in_process/run_in_process.py

```diff
@@ -128,67 +94,126 @@ async def _monitor_state(
 @trio_typing.takes_callable_and_args
 async def open_in_process(
     async_fn: Callable[..., TReturn], *args: Any
-) -> AsyncIterator[Process[TReturn]]:
-    proc = Process(async_fn, args)
-
-    parent_r, child_w = os.pipe()
-    child_r, parent_w = os.pipe()
-    parent_pid = os.getpid()
-
-    command = get_subprocess_command(child_r, child_w, parent_pid)
+) -> AsyncIterator[ProcessAPI[TReturn]]:
+    async with open_worker_process() as worker:
+        async with worker.open(async_fn, *args) as proc:
+            yield proc
 
-    sub_proc = await trio.open_process(
-        command,
-        # stdin=subprocess.PIPE,
-        # stdout=subprocess.PIPE,
-        # stderr=subprocess.PIPE,
-        pass_fds=(child_r, child_w),
-    )
 
-    async with trio.open_nursery() as nursery:
-        nursery.start_soon(_monitor_sub_proc, proc, sub_proc, parent_w)
-
-        async with trio.hazmat.FdStream(parent_r) as from_child:
+@trio_typing.takes_callable_and_args
+async def run_in_process(async_fn: Callable[..., TReturn], *args: Any) -> TReturn:
+    async with open_in_process(async_fn, *args) as proc:
+        await proc.wait()
+    return proc.get_result_or_raise()
+
+
+class WorkerProcess(WorkerProcessAPI):
+    def __init__(
+        self,
+        trio_process: trio.Process,
+        from_child: trio.lowlevel.FdStream,
+        to_child: trio.lowlevel.FdStream,
+    ) -> None:
+        self._trio_proc = trio_process
+        self._from_child = from_child
+        self._to_child = to_child
+        self._busy = False
+        self._dead = False
+
+    # Need the type:ignore comment because mypy complains this method's signature is incompatible
+    # with its supertype.
```

Comment:
see https://github.com/ethereum/ddht/pull/261#discussion_r539417748

You just need to drop the `async` on the ABC implementation.

Reply:
Cool!